src/Route.ts

```diff
@@ -45,26 +45,33 @@ export type Route<
   headersDecoder: Decoder<Headers>
 }
 
+type CombineFn<
```

Comment:
Not certain about this name

Reply:
This is really nice. Probably `CombineRoute`, since it is specialised to that.